### ISSUE TITLE
TriggerBinding doesn't support multiline string param?

### ISSUE THREAD

**Issue by gejunqiang** [tektoncd/triggers#772]
# Problem Description

I want to trigger my TaskRun dynamically with Tekton Triggers. And in TriggerBinding, I use a param named dockerfile, and the value is a constant  multiline string, it's shown as follows:

```yaml
apiVersion: triggers.tekton.dev/v1alpha1
kind: TriggerBinding
metadata:
  name: poc-taskbinding
  namespace: poc
spec:
  params:
    - name: gitrevision
      value: $(body.git.revision)
    - name: gitrepositoryurl
      value: $(body.git.ssh_url)
    - name: imageurl
      value: $(body.image.url)
    - name: imagetag
      value: $(body.image.tag)
    - name: dockerfile
      value: |
        FROM alpine:3.9
        ADD artifacts.tar.gz /artifacts/
```

Then I use this command to call the event listener:

```bash
curl -v -X POST poc-trigger.gjq.netease.com -d '
{
  "git": {
    "revision": "master",
    "ssh_url": "git@github.com:gejunqiang/test-deploy-javawebapp.git"
  },
  "image":{
    "url": "hub.c.163.com/gejunqiang/poc-javawebapp",
    "tag": "v1"
  }
}'
```

But it doesn't work. And the event listener pod logs messages with a level: error, the logs are shown as below:

```
{"level":"info","logger":"eventlistener","caller":"sink/sink.go:181","msg":"ResolvedParams : [{Name:gitrepositoryurl Value:git@github.com:gejunqiang/test-deploy-javawebapp.git} {Name:imageurl Value:hub.c.163.com/gejunqiang/poc-javawebapp} {Name:imagetag Value:v1} {Name:dockerfile Value:FROM alpine:3.9\nADD artifacts.tar.gz /artifacts/\n} {Name:gitrevision Value:master}]","knative.dev/controller":"eventlistener","/triggers-eventid":"njg45","/trigger":""}
{"level":"error","logger":"eventlistener","caller":"sink/sink.go:266","msg":"problem creating obj: &errors.errorString{s:\"couldn't unmarshal json: invalid character '\\\\n' in string literal\"}","knative.dev/controller":"eventlistener","/triggers-eventid":"njg45","/trigger":"","stacktrace":"github.com/tektoncd/triggers/pkg/sink.Sink.CreateResources\n\tgithub.com/tektoncd/triggers/pkg/sink/sink.go:266\ngithub.com/tektoncd/triggers/pkg/sink.Sink.processTrigger\n\tgithub.com/tektoncd/triggers/pkg/sink/sink.go:183\ngithub.com/tektoncd/triggers/pkg/sink.Sink.HandleEvent.func1\n\tgithub.com/tektoncd/triggers/pkg/sink/sink.go:94"}
{"level":"error","logger":"eventlistener","caller":"sink/sink.go:184","msg":"couldn't unmarshal json: invalid character '\\n' in string literal","knative.dev/controller":"eventlistener","/triggers-eventid":"njg45","/trigger":"","stacktrace":"github.com/tektoncd/triggers/pkg/sink.Sink.processTrigger\n\tgithub.com/tektoncd/triggers/pkg/sink/sink.go:184\ngithub.com/tektoncd/triggers/pkg/sink.Sink.HandleEvent.func1\n\tgithub.com/tektoncd/triggers/pkg/sink/sink.go:94"}
```

From the first line of logs, we can see that the TriggerBinding resolved params correctly. Then it caused an error: `&errors.errorString{s:\"couldn't unmarshal json: invalid character '\\\\n' in string literal\"}"`. I guess maybe it's the problem with newline character in multiline string, Or maybe  the way I use the TriggerBinding is wrong. Can someone give me some suggestions.

# Additional Info
The version of Tekton Triggers: v0.8.1

**Comment by dibyom** [comment]
The error location looks very similar to #777

**Comment by dibyom** [comment]
This should have been fixed with #823 
Please feel free to reopen if this is still an issue!
/close
